sycl/doc/extensions/DiscardQueueEvents/SYCL_EXT_ONEAPI_DISCARD_QUEUE_EVENTS.asciidoc

```diff
@@ -0,0 +1,223 @@
+|===
+|Value |Description
+|1     |Initial extension version.  Base features are supported.
+|===
+
+== Overview
+
+This extension adds `ext::oneapi::property::queue::discard_queue_events` property for `sycl::queue`,
+by using this property the application informs a SYCL implementation that the returned event can be discarded.
+
+In cases where it will be possible, the implementation makes a decision about internal optimization -- avoiding
+the creation of low-level event from the backend. The property affects only kernel cases,
+i.e. avoid creating low-level event for submission of kernels.
+Performance improvement is the main motivation for using this property.
+
+Below is a usage example:
+[source,c++]
+----
+  sycl::property_list props{ext::oneapi::property::queue::discard_queue_events{}
+                            property::queue::in_order{}};
+  sycl::queue Queue( props );
+
+  // some USM preparations ..
+
+  sycl::event e1, e2, e3;
+
+  // returning "invalid" events from each submission function:
+  e1 = Queue.parallel_for(NDRange, [=](nd_item<1> item){ do_smth1(); });
+
+  e2 = Queue.single_task([=](){ do_smth2(); });
+
+  e3 = Queue.submit([&](handler &CGH) { CGH.parallel_for(NDRange, [=](nd_item<1> item){ do_smth3(); }); });
+
+  Queue.wait();
+----
+
+In the example above, the application doesn't use sycl events: `e1`, `e2`, `e3`
+and is waiting for the end of work by `queue::wait()`. The returned events will be
+_invalid_ events. _invalid_ event is a `sycl::event` with a status `invalid` and the event returns from each kernel
+submission when queue is created with the `discard_queue_events` property.
+See the description of behavior for this event below for details.
+
+_invalid_ event API behavior:
+[source,c++]
+----
+// the behavior is the same as for the default event, i.e. must throw an exception with the errc::invalid_object error code
+cl_event get() const
+
+// the behavior is the same as for the default event, i.e. must return true
+bool is_host() const;
+
+// the behavior is the same as for the default event, i.e. must return backend::host
+backend get_backend() const noexcept;
+
+// the behavior is the same as for the default event, i.e. must return a zero-sized vector
+std::vector<event> get_wait_list();
+
+// must throw an exception with the errc::invalid_object error code.
+void wait();
+
+// passing invalid event in the function must throw an exception with the errc::invalid_object error code.
+static void wait(const std::vector<event> &eventList);
+
+// must throw an exception with the errc::invalid_object error code.
+void wait_and_throw();
+
+// passing invalid event in the function must throw an exception with the errc::invalid_object error code.
+static void wait_and_throw(const std::vector<event> &eventList);
+
+// returns info::event_command_status::ext_oneapi_unknown status
+get_info<info::event::command_execution_status>() const;
+
+// the behavior is the same as for the default event, i.e. returns 0

```

Comment:
Hm, does default event really returns `0`? Is it defined by the spec? If no suggest removing `i.e. returns 0`.

Reply:
Yes, the default event returns 0, but I cannot find the exact statement of that in spec. So I will remove it. Thanks.